pyk/src/pyk/kcfg/minimize.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import annotations
+
+from functools import reduce
+from typing import TYPE_CHECKING
+
+from pyk.cterm import CTerm
+from pyk.utils import not_none, single
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from pyk.kcfg.kcfg import KCFG, NodeIdLike
+
+
+class KCFGMinimizer:
```

Comment:
Just to confirm: does `KCFG` have any private attributes this class needs access to?

Reply:
self.kcfg._resolve(b_id) at L134

Reply:
Does it matter?

Reply:
> Does it matter?

In general, yes.
* There's no guarantee for backwards compatibiliy on a private method.
* A class can only guarantee its invariants if it's internals are not tinkered with.

However, this particular case is pretty innocuous. Still, it'd great to eliminate it (e.g. by `node(b_id).id`).

Reply:
Thank you for your suggestion! I’ll make that change. But I have a question: will `node(b_id).id` be significantly slower than the current implementation?

Reply:
It's not significantly slower, it's just one extra map lookup plus an attribute access.